build/Targets/Settings.targets

```diff
@@ -48,6 +48,16 @@
   <Import Project="$(MSBuildExtensionsPath)\Microsoft\NuGet\Microsoft.NuGet.props"
           Condition="'$(OS)' != 'Windows_NT'" />
 
+  <!-- The /publicsign argument is required for the compiler, but the current MSBuild
+       build task can't be redirected and even if it could the new build task is not
+       built against the xplat MSBuild references so it can't be loaded properly. Providing
+       a response file works around the problem by directly adding the public sign argument
+       to all unix compilations. This shouldn't present a problem as it's impossible to
+       fully sign on unix at the moment anyway -->
```

Comment:
Yeah, this is kind of nasty. I've filed https://github.com/dotnet/roslyn/issues/7756 to track removing this and I'll add a mention of the bug number in the comment.

There are some underlying infrastructrue issues here. First, it appears the MSBuild on Mono eagerly loads tasks listed in Microsoft.Common.tasks. That means that the default task is loaded before we have a chance to redirect it to anywhere else.

But even if that were fixed (which would be difficult, I think, because I don't think the MSBuild version for Mono is getting any updates) there would still be another issue -- our build task is currently built against the Windows desktop MSBuild references. Theoretically, we could dig through the Mono MSBuild NuGet package, find the references, and build a *third* build task against those references.

I think that's getting a bit ridiculous though -- we already have a build task built against the CoreCLR MSBuild references that's actively being developed. We should instead start moving forward to using that. I've filed bug https://github.com/dotnet/roslyn/issues/7755 for that.